lib/ex_uid2/api/request.ex

```diff
@@ -0,0 +1,39 @@
+defmodule ExUid2.Api.Request do
+
+  def make_encrypted_request_body(payload, secret_key, ts) do
+    with {:unencryted_envelope, {:ok, envelope}} <- {:unencryted_envelope, make_unencrypted_envelope(payload, ts)},
+         {:encrypted_envelope, {:ok, encrypted_envelope}} <- {:encrypted_envelope, encrypt_envelope(envelope, secret_key)} do
+           {:ok, encrypted_envelope}
+    else
+      {_, error} -> error
+    end
+  end
+
+  def make_unencrypted_envelope(payload, ts, nonce \\ :crypto.strong_rand_bytes(8))
+  def make_unencrypted_envelope(payload, ts, nonce)
+      when is_binary(payload) and byte_size(nonce) == 8 and is_integer(ts) do
+    ts_bin = :binary.encode_unsigned(ts, :big)
+    pad = 8 - byte_size(ts_bin)
+    encoded_ts = <<0::pad*8, ts_bin::binary>>
```

Comment:
I feel like this is a little sketchy. Do they truly want a 16 byte timestamp? Also `encode_unsigned` outputs the smallest binary representation of the provided integer so if someone provides the ts 2, then you'll only end up with 9 bytes here instead of 16.

Reply:
I completely misread the code...